core/designsystem/src/main/java/com/hilingual/core/designsystem/component/textfield/HilingualBasicTextField.kt

```diff
@@ -0,0 +1,135 @@
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.text.BasicTextField
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.focus.focusRequester
+import androidx.compose.ui.focus.onFocusChanged
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.unit.dp
+import com.hilingual.core.designsystem.theme.HilingualTheme
+
+@Composable
+fun HilingualBasicTextField(
+    value: String,
+    placeholder: String,
+    textStyle: TextStyle,
+    onValueChanged: (String) -> Unit,
+    onFocusChanged: (Boolean) -> Unit,
+    modifier: Modifier = Modifier,
+    borderModifier: Modifier = Modifier,
+    backgroundColor: Color = HilingualTheme.colors.gray100,
+    focusRequester: FocusRequester = FocusRequester(),
+    singleLine: Boolean = true,  // 한 줄 입력만 허용할지 여부
+    leadingIcon: @Composable (() -> Unit)? = null,
+    trailingIcon: @Composable (() -> Unit)? = null,
+    keyboardImeAction: ImeAction = ImeAction.Done,  // 키보드 오른쪽 아래 버튼 (기본값: Done)
+    onDoneAction: () -> Unit = {},  // 키보드의 완료 액션이 눌렸을 때 실행되는 콜백
+    onSearchAction: () -> Unit = {},  // 키보드의 검색 액션이 눌렸을 때 실행되는 콜백
+    bottomRightContent: @Composable (() -> Unit)? = null
+) {
+    BasicTextField(
+        value = value,
+        onValueChange = onValueChanged,
+        modifier = modifier
+            .clip(RoundedCornerShape(8.dp))
+            .background(backgroundColor)
+            .then(borderModifier)
+            .padding(horizontal = 12.dp)
+            .focusRequester(focusRequester)
+            .onFocusChanged { focusState ->
+                onFocusChanged(focusState.isFocused)
+            },
+        singleLine = singleLine,
+        keyboardOptions = KeyboardOptions(imeAction = keyboardImeAction),
+        keyboardActions = KeyboardActions(
+            onDone = { onDoneAction() },
+            onSearch = { onSearchAction() }
+        ),
+        textStyle = textStyle.copy(
+            color = HilingualTheme.colors.black
+        ),
+        decorationBox = { innerTextField ->
+            Box {
+                val columnModifier = if (bottomRightContent == null) {
+                    Modifier.fillMaxSize()
+                } else {
+                    Modifier
+                        .fillMaxSize()
+                        .padding(top = 12.dp, bottom = 39.dp)
+                }
```

Comment:
p3
이거 저 합세 하다가 저희 리드 분이 사용하셨던 건데,
```
fun Modifier.ignoreNextModifiers(): Modifier {
    return object : Modifier by this {

        override fun then(other: Modifier): Modifier {
            return this
        }
    }
}
```
식으로 확장 함수 만들어두면
```suggestion
                val columnModifier = Modifier
                    .fillMaxSize()
                    .let { if (bottomRightContent == null) it else it.ignoreNextModifiers() }
```
이렇게도 사용할 수 있어요!
지금은 사용할만한 곳이 여기 외엔 크게 없는 것 같아서 '이런 것도 있다~' 정도로 남겨봐용

Reply:
요것도 좋지만 .then이라는 키워드가 있습니다 이걸로 적용해도 좋아요. 지금 구현은 Modifier객체를 재생성하게 댈거에요

Reply:
@nahy-512 `.let` 이 편리하긴 하지만 내부적으로 생성되는 코드 양의 차이가 있습니다ㅎㅎ 그래서 단순 null체킹은 if로 해주는게 좋아요

Reply:
오호.. 위에서 지영 언니가 then 사용해준 걸 봤었는데 여기도 활용할 수 있겠군요.
덕분에 하나 더 배워갑니다☺️

Reply:
해당 확장 함수나 `.let`과 `if-else` 분기 처리의 차이점은 저도 이번에 첨 알게 되었네요..! 덕분에 배워갑니당 💪🏻